src/test/java/spoon/test/template/SubstitutionTest.java

```diff
@@ -0,0 +1,60 @@
+package spoon.test.template;
+
+import org.junit.jupiter.api.Test;
+import spoon.Launcher;
+import spoon.reflect.code.CtExpression;
+import spoon.reflect.declaration.CtField;
+import spoon.reflect.declaration.CtType;
+import spoon.reflect.factory.Factory;
+import spoon.support.compiler.FileSystemFile;
+import spoon.template.StatementTemplate;
+import spoon.template.Substitution;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class SubstitutionTest {
+
+    public class SingleFieldTemplate extends StatementTemplate {
+        String testString = "goodName";
+
+        @Override
+        public void statement() { }
+    }
```

Comment:
This is a personal preference of mine, and others may think otherwise, but I would put this _below_ the test. I like to read files top-down, so if a method A uses a method or class B which is defined in the same file, then A is defined before B.

Again, I think this is a matter of taste, and you should form your own opinion. If you agree, then move the class down. If you disagree, feel free to keep it where it is.

Reply:
I will place it downwards with no issues, actually, the first programming language that I learnt was C++, and in that one should declare the function at the top of the file calling the function, so probably I have this habit from there.

Reply:
Totally understand, and as I said it's a matter of taste. Some prefer declaring functions before they are used, while others like me prefer to declare them just after they are used. There's no right or wrong, but in Java I find that the latter is more common.